doctest/doctest.h

```diff
@@ -137,67 +145,74 @@
+    DOCTEST_GCC_SUPPRESS_WARNING("-Weffc++")                                                       \
+    DOCTEST_GCC_SUPPRESS_WARNING("-Wstrict-overflow")                                              \
+    DOCTEST_GCC_SUPPRESS_WARNING("-Wstrict-aliasing")                                              \
+    DOCTEST_GCC_SUPPRESS_WARNING("-Wmissing-declarations")                                         \
+    DOCTEST_GCC_SUPPRESS_WARNING("-Wunused-local-typedefs")                                        \
+    DOCTEST_GCC_SUPPRESS_WARNING("-Wuseless-cast")                                                 \
+    DOCTEST_GCC_SUPPRESS_WARNING("-Wnoexcept")                                                     \
+                                                                                                   \
+    DOCTEST_MSVC_SUPPRESS_WARNING_PUSH                                                             \
+    /* these 4 also disabled globally via cmake: */                                                \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4514)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4571)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4710)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4711)                                                            \
+    /* */                                                                                          \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4616)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4619)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4996)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4706)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4512)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4127)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4820)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4626)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(5026)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(5027)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4625)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(4640)                                                            \
+    DOCTEST_MSVC_SUPPRESS_WARNING(5045)                                                            \
+    /* static analysis */                                                                          \
```

Comment:
I think all MSVC warnings could keep their comments for what they are for with such comments in the multiline macro - I think it's pretty useful for future reference and would avoid needing to google numbers